libsyclinterface/CMakeLists.txt

```diff
@@ -60,20 +64,28 @@ if(DPCTL_ENABLE_L0_PROGRAM_CREATION)
         get_level_zero_headers()
     endif()
     if (UNIX)
-        find_library(PI_LEVEL_ZERO_LIB
-            NAMES pi_level_zero ur_adapter_level_zero
-            HINTS ${IntelSyclCompiler_LIBRARY_DIR}
-        )
-        find_program(READELF_PROG readelf)
-        find_program(GREP_PROG grep)
-        execute_process(
-            COMMAND ${READELF_PROG} -d ${PI_LEVEL_ZERO_LIB}
-            COMMAND ${GREP_PROG} libze_loader
-            COMMAND ${GREP_PROG} -Po "libze_loader[^\]]*"
-            OUTPUT_VARIABLE LIBZE_LOADER_FILENAME
-            OUTPUT_STRIP_TRAILING_WHITESPACE
-            ERROR_STRIP_TRAILING_WHITESPACE
-        )
+        if ("x${LIBZE_SET_LOADER_FILENAME}" STREQUAL "x")
+            find_library(PI_LEVEL_ZERO_LIB
+                NAMES pi_level_zero ur_adapter_level_zero
+                HINTS ${IntelSyclCompiler_LIBRARY_DIR}
+            )
+            find_program(READELF_PROG readelf)
+            find_program(GREP_PROG grep)
+            execute_process(
+                COMMAND ${READELF_PROG} -d ${PI_LEVEL_ZERO_LIB}
+                COMMAND ${GREP_PROG} libze_loader
+                COMMAND ${GREP_PROG} -Po "libze_loader[^\]]*"
+                OUTPUT_VARIABLE LIBZE_LOADER_FILENAME
+                OUTPUT_STRIP_TRAILING_WHITESPACE
+                ERROR_STRIP_TRAILING_WHITESPACE
+            )
+            # if libze_loader is statically linked, LIBZE_LOADER_FILENAME will be an empty string
+            if ("x${LIBZE_LOADER_FILENAME}" STREQUAL "x")
+                set(LIBZE_LOADER_FILENAME ${LIBZE_DEFAULT_LOADER_FILENAME})
+            endif()
+        else ()
+            set(LIBZE_LOADER_FILENAME ${LIBZE_SET_LOADER_FILENAME})
+        endif()
         set(LIBZE_LOADER_FILENAME ${LIBZE_LOADER_FILENAME} PARENT_SCOPE)
```

Comment:
Should we remove that line? It seems duplicating with [one](https://github.com/IntelPython/dpctl/pull/2025/files#diff-df76a50a4c90962d8ff51b821a9f862a80054464e8880eda6aa648fa5bcbdc80R91) below.

Reply:
Difficult to say, it does duplicate, but it was already present in the code before, so I would want to be sure that it never served a purpose

Reply:
I don't mind to handle that separately if needed.